tests/functional/local-overlay-store/verify-inner.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail
+
+set -x
+
+source common.sh
+
+# Avoid store dir being inside sandbox build-dir
+unset NIX_STORE_DIR
+unset NIX_STATE_DIR
+
+storeDirs
+
+initLowerStore
+
+mountOverlayfs
+
+
+## Initialise stores for test
+
+# Realise a derivation from the lower store to propagate paths to overlay DB
+nix-store --store "$storeB" --realise $drvPath
+
+# Also ensure dummy file exists in overlay DB
+dummyPath=$(nix-store --store "$storeB" --add ../dummy)
+
+# Add something to the lower store that will not be propagated to overlay DB
+lowerOnlyPath=$(addTextToStore "$storeA" lower-only "Only in lower store")
+
+# Verify should be successful at this point
+nix-store --store "$storeB" --verify --check-contents
+
+# Make a backup so we can repair later
+backupStore="$storeVolume/backup"
+mkdir "$backupStore"
+tar -cC "$storeBRoot" nix | tar -xC "$backupStore"
```

Comment:
(nit): Any reason for the tarpipe here rather than a `cp -r`?

Reply:
@benradf tells me was being cautious about preserving permissions, but then he suggestion `-a` with `cp`, and that seems to work just fine!